211018_4win_main.py

```diff
@@ -49,28 +49,21 @@ def digonal_win_left_top(player):
 
 def diagonal_win_left_bottom(player):
     count = 0
-    for column in range(6):
-        n = -2
-        for row in range(6):
-            if column + n == 0:
-                count = 0
-            try:
-                if matrix[row][column + n] == player:
-                    print("+")
-                    count += 1
+    for column in range(7):
+        for row in range(5, 0 - 1, -1):             # begin to look after a player token on position "50"
+            if matrix[row][column] == player:
+                for token in range(4):             # begin to check, if the token begins a connect4
+                    if row < 3 or column > 3:       # on position "33": last chance to begin a connect4 from left bottom to right up 
+                        return True
+                    if matrix[row - token][column + token] == player:   # check if their are other player tokens on the up right diagonal
+                        count += 1

```

Comment:
Nice Olli,
I like the appoach.
It took me some mind bending and Im confinced that there is even shorter version out there, but for now: I agree with your changes.
They are way less hacky than the n-variable I convinced Leo to use